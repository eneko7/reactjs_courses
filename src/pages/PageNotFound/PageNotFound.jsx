import React from 'react';
import Header from '../../components/Header';
// import ResultLine from '../../components/ResultLine';
// import MoviesGrid from '../../components/MoviesGrid';
import Footer from '../../components/Footer';
import ErrorBoundary from '../../components/ErrorBoundary/ErrorBoundary';
import styles from './PageNotFound.scss';

const PageNotFound = () => (
  <ErrorBoundary>
    <Header />
    <article className={styles.page_not_found_wrap}>
      <span className={styles.page_not_found_wrap__title}>404 ERROR</span>
      <span className={styles.page_not_found_wrap__title}>Page Not Found</span>
    </article>
    <Footer />
  </ErrorBoundary>
);

export default PageNotFound;
