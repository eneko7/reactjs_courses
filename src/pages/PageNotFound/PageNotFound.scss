@import "../../assets/constants/constants.scss";

.page_not_found_wrap {
  padding: 3% 15%;
  background-color: #f3efef96;
  display: flex;
  align-items: center;
  align-content: center;
  flex-wrap: wrap;
  min-height: 48vh;

  .page_not_found_wrap__title {
    width: 100%;
    text-align: center;
    font-size: 40px;
  }

  .page_not_found_wrap__title:first-child {
    color: $roseColor;
  }
}

@media (max-width: $fulhdDesktop) {
  .page_not_found_wrap {
    padding: 1% $leftPaddingLaptop;
  }
}

@media (max-width: $laptop) {
  .page_not_found_wrap {
    padding: 3% $leftPaddingTablet;
  }
}

@media (max-width: $tablet) {
  .page_not_found_wrap {
    padding: 5% 5%;
  }
}
