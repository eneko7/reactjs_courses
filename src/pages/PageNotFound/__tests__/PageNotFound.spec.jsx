import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import PageNotFound from '../PageNotFound';

const renderer = new ShallowRenderer();
describe('PageNotFound', () => {
  it('renders correctly', () => {
    renderer.render(<PageNotFound />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
