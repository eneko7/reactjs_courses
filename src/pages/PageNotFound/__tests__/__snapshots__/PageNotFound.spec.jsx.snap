// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound renders correctly 1`] = `
<ErrorBoundary>
  <withRouter(Connect(Header)) />
  <article
    className="page_not_found_wrap"
  >
    <span
      className="page_not_found_wrap__title"
    >
      404 ERROR
    </span>
    <span
      className="page_not_found_wrap__title"
    >
      Page Not Found
    </span>
  </article>
  <Footer />
</ErrorBoundary>
`;
